### PR TITLE
Fix Issue 337

### PR DESCRIPTION
**Pull request by m-chaturvedi**
[Similar cpython issue](https://bugs.python.org/issue4474)

Towards #337

### CODE REVIEW

**Comment by m-chaturvedi** [comment]
I signed it!

File: cpplint/cpplint.py [hunk 1]
```diff
@@ -4291,6 +4292,15 @@ def GetLineWidth(line):
       if unicodedata.east_asian_width(uc) in ('W', 'F'):
         width += 2
       elif not unicodedata.combining(uc):
+        # Issue 337        
+        if (sys.version_info.major, sys.version_info.minor) == (2, 7) :
+          # https://github.com/python/cpython/blob/2.7/Include/unicodeobject.h#L81
+          is_wide_build = sysconfig.get_config_var("Py_UNICODE_SIZE") >= 4
+          # https://github.com/python/cpython/blob/2.7/Objects/unicodeobject.c#L564
+          is_low_surrogate = (0xDC00 <= ord(uc) and ord(uc) <= 0xDFFF)
```
Comment (gpshead) [inline cpplint/cpplint.py]: I believe this can be written as `0xDC00 <= ord(uc) <= 0xDFFF`

File: cpplint/cpplint.py [hunk 2]
```diff
@@ -4291,6 +4292,15 @@ def GetLineWidth(line):
       if unicodedata.east_asian_width(uc) in ('W', 'F'):
         width += 2
       elif not unicodedata.combining(uc):
+        # Issue 337        
+        if (sys.version_info.major, sys.version_info.minor) == (2, 7) :
```
Comment (gpshead) [inline cpplint/cpplint.py]: I suggest writing this as `if sys.version_info <= (2,):` as this is not == 2.7 specific.  technically `<= (3,2)` makes sense but we're never going to support the long obsolete early 3.0-3.2 interpreters.

**Review by gpshead — COMMENTED** [review]
is there a meaningful way to unittest this in cpplint_unittest.py?  (already there, or does one need to be added?)

Even a test that if run on both unicode width builds of cpython 2.7 interpreter would at least execute this code path to demonstrate the problem if this code weren't here would be good.

**Comment by m-chaturvedi** [comment]
I don't follow this:

>Even a test that if run on both unicode width builds of cpython 2.7 interpreter would at least execute this code path to demonstrate the problem if this code weren't here would be good.

I added a test which will fail on UCS2 Python builds without the proposed change in `cpplint.py`.  Would that be sufficient?

File: cpplint/cpplint.py (hunk 1 above)
Comment (m-chaturvedi) [reply cpplint/cpplint.py]: Thanks!

File: cpplint/cpplint.py (hunk 2 above)
Comment (m-chaturvedi) [reply cpplint/cpplint.py]: Thanks!

**Comment by m-chaturvedi** [comment]
Mac builds seem to be `ucs2`, btw, which is where this problem got exposed to us.

**Comment by m-chaturvedi** [comment]
@gpshead Just a gentle reminder!  Thanks!

**Comment by gpshead** [comment]
thanks, yes a test that will fail on such a built is good enough.

**Pull request merged**